addons/account/models/account_move.py

```diff
@@ -4541,8 +4575,6 @@ def create(self, vals_list):
         moves = lines.mapped('move_id')
         if self._context.get('check_move_validity', True):
             moves._check_balanced()
-        moves._check_fiscalyear_lock_date()
-        lines._check_tax_lock_date()
```

Comment:
We need to be sure that we can't create new lines for non draft moves though

Reply:
Checked with TSB, this can happen when adding write-off lines during reconciliation.
I'm not sure that it is relevant here, but to be safe we can keep the check for posted entries.